core/services/backend_monitor.go

```diff
@@ -15,22 +15,22 @@ import (
 	gopsutil "github.com/shirou/gopsutil/v3/process"
 )
 
-type BackendMonitor struct {
+type BackendMonitorService struct {
 	configLoader *config.BackendConfigLoader

```

Comment:
because we have "configs" in multiple places, can we name this "backendConfigLoader" to make it a bit more specific? Will help to reduce confusion and cognitive load remembering what everything is throughout the codebase.